cmd/serve.go

```diff
@@ -424,6 +448,21 @@ func (c *ServeCommand) Command(serverService ServerServiceInterface, monitorServ
 			}
 			serveOpts.AnchorPlatformAPIService = apAPIService
 
+			// Kafka (background)
+			kafkaEventManager, err := di.NewKafkaEventManager(ctx, serveOpts.Brokers, serveOpts.Topics, serveOpts.ConsumerGroupID, &events.PingPongEventHandler{})
```

Comment:
I believe we should separate `KafkaConsumer` and `KafkaProducer`. `KafkaEventManager` seems to mix both concepts even though they're pretty separate. 
## KafkaProducer: 
- requires just brokers and topic. 
- is going to be used all over the code, whenever we need to produce a message. 
- lightweight. 
- we only need one instance. 

## KafkaConsumer
- we need multiple instances 
- it only needs to be defined on startup. 
- heavyweight (requires looping). 

This is just example to illustrate what I'm thinking 
```go
// Kafka Producer.
serveOpts.KafkaProducer, err := di.NewKafkaProducer(ctx, serveOpts.Brokers)
// error handling
defer serveOpts.KafkaProducer.Close()

// Kafka Consumers
// 1. Consumer for `sdp.payments.payments_submitted` topic
paymentsConsumer, err :=  di.NewKafkaConsumer(ctx, serveOpts.Brokers, serveOpts.TopicPaymentSubmitted, serveOpts.ConsumerGroupID, &handlers.SubmittedPaymentEventHandler{})
// error handling
defer paymentSubmittedConsumer.Close()

go func() {
   err := events.Consume(ctx, paymentSubmittedConsumer)
}

// 2. Consumer for `sdp.payments.payment_status_updated` topic
paymentStatusUpdatedConsumer, err :=  di.NewKafkaConsumer(ctx, serveOpts.Brokers, serveOpts.TopicPaymentStatusUpdated, serveOpts.ConsumerGroupID,
&handlers.PaymentStatusUpdatedEventHandler{})
// error handling
defer paymentStatusUpdatedConsumer.Close()

go func() {
    err := events.Consume(ctx, paymentStatusUpdatedConsumer)
}
```

We can talk about it before you make the change if you think this won't work.

Reply:
changed: [87c250d](https://github.com/stellar/stellar-disbursement-platform-backend/pull/119/commits/87c250d9d3feb26893f037c7735c336056f96743)